test/cardTests/debitCardTests.java

```diff
@@ -0,0 +1,1084 @@
+
+			Address newBillingAddress = new Address("6645 wash ave", "STL", "MO", "63130");
+			String newBillingAddressString = newBillingAddress.getFullAddress();
+			boolean result = testDebitCard.setBillingAddress(newBillingAddress, databaseConnection);
+			assertTrue(result, "setting billing address should have returned true if successful");
+
+			DebitCardEntity queriedCardEntity = debitCardDao.queryForId(Encryption.encrypt(cardNumber));
+			assertFalse(queriedCardEntity == null, "card should be found in database");
+
+			DebitCard queriedCard = new DebitCard(queriedCardEntity);
+			String queriedBillingAddress = queriedCard.getBillingAddress();
+			assertTrue(queriedBillingAddress.equals(newBillingAddressString), "Building Address was not updated. Expected: " + newBillingAddressString + " but got: " + queriedBillingAddress);
+
+			Map<String, Object> addressQueryParams = new HashMap<String, Object>();
+			addressQueryParams.put("street_address", Encryption.encrypt("444 E 82nd st"));
+			addressQueryParams.put("city", Encryption.encrypt("NYC"));
+			addressQueryParams.put("state", Encryption.encrypt("NY"));
+			addressQueryParams.put("zip_code", Encryption.encrypt("10025"));
+
+			List<AddressEntity> returnedAddresses = addressDao.queryForFieldValues(addressQueryParams);
+			assertTrue(returnedAddresses.size() == 1, "Should not have deleted the old address since it is associated with another card");
+
+			Map<String, Object> addressQueryParamsTwo = new HashMap<String, Object>();
+			addressQueryParamsTwo.put("street_address", Encryption.encrypt("6645 wash ave"));
+			addressQueryParamsTwo.put("city", Encryption.encrypt("STL"));
+			addressQueryParamsTwo.put("state", Encryption.encrypt("MO"));
+			addressQueryParamsTwo.put("zip_code", Encryption.encrypt("63130"));
+
+			List<AddressEntity> returnedAddressesTwo = addressDao.queryForFieldValues(addressQueryParamsTwo);
+			assertTrue(returnedAddressesTwo.size() == 1, "Should have added another row for new address");
+
+			debitCardDao.delete(testDebitCardEntity);
+			debitCardDao.delete(testDebitCardEntityTwo);
+			newBillingAddress.deleteAddress(databaseConnection);
+			oldBillingAddress.deleteAddress(databaseConnection);
+		} catch (SQLException e) {
+			e.printStackTrace();
+			fail(e);
+		}
+	}
+
+	@Test
+	void testUpdateDebitCardInformationUpdateNickname() {
+		String nickname = "nick";
+		String cardNumber = "92347109485193857";
+		String expirationDate = "04/23";
+		String cvv = "988";
+		String pin ="6789";
+		try {
+			DebitCard testDebitCard = new DebitCard(testUser, nickname, cardNumber, expirationDate, cvv,pin, testAddress);
+			DebitCardEntity testDebitCardEntity = testDebitCard.getDebitCardEntity();
+			Dao<DebitCardEntity, String> debitCardDao = DaoManager.createDao(databaseConnection, DebitCardEntity.class);
+			debitCardDao.create(testDebitCardEntity);
+
+			try {
+				File userInput = new File("test/cardTests/updateCreditCardNicknameInput.txt");
```

Comment:
any files that are used for both credit and debit maybe we should rename to just like "updateCardNicknameInput.txt" but then would need to fix in credit card as well